cli/script/command-executor.ts

```diff
@@ -926,17 +926,28 @@ export var release = (command: cli.IReleaseCommand): Promise<void> => {
     var uploadProgress = (currentProgress: number): void => {
         progressBar.tick(currentProgress - lastTotalProgress);
         lastTotalProgress = currentProgress;
-    }
+    };
 
+    var deleteFileIfTemporary = (file: IPackageFile): void => {
+        if (file.isTemporary) {
+            fs.unlinkSync(filePath);
+        }
+    };
+    
     return getPackageFilePromise
         .then((file: IPackageFile): Promise<void> => {
             return sdk.release(command.appName, command.deploymentName, file.path, command.appStoreVersion, command.description, command.mandatory, uploadProgress)
                 .then((): void => {
                     log("Successfully released an update containing the \"" + command.package + "\" " + (isSingleFilePackage ? "file" : "directory") + " to the \"" + command.deploymentName + "\" deployment of the \"" + command.appName + "\" app.");
 
-                    if (file.isTemporary) {
-                        fs.unlinkSync(filePath);
-                    }
+                    deleteFileIfTemporary(file);
+                })
+                .catch((error): void => {
+                    deleteFileIfTemporary(file);
```

Comment:
I was going to, but I decided to mirror what you did in the `release-react` command when deleting the temp folder. Since `Promise.prototype.finally` isn't standard, I kind of prefer doing this, but I'm happy either way.